src/raftstore/store/peer.rs

```diff
@@ -866,58 +963,71 @@ impl Peer {
         self.apply_scheduler.schedule(t).unwrap();
 
         self.proposals.push(meta);
-
-        true
     }
 
-    fn should_read_local(&mut self, req: &RaftCmdRequest) -> bool {
+    fn get_handle_policy(&mut self, req: &RaftCmdRequest) -> Result<RequestPolicy> {
+        if req.has_admin_request() {
+            if apply::get_change_peer_cmd(req).is_some() {
+                return Ok(RequestPolicy::ProposeConfChange);
+            }
+            if get_transfer_leader_cmd(req).is_some() {
+                return Ok(RequestPolicy::ProposeTransferLeader);
+            }
+            return Ok(RequestPolicy::ProposeNormal);
+        }
+
+        let mut is_read = false;
+        let mut is_write = false;
+        for r in req.get_requests() {
+            match r.get_cmd_type() {
+                CmdType::Get | CmdType::Snap => is_read = true,
+                CmdType::Delete | CmdType::Put => is_write = true,
+                CmdType::Invalid => {
+                    return Err(box_err!("invalid cmd type, message {:?} maybe currupted",
+                                        req.get_header().get_uuid()));
+                }
+            }
+
+            if is_read && is_write {
+                return Err(box_err!("read and write can't be mixed in one batch."));
+            }
+        }
+
+        if is_write {
+            return Ok(RequestPolicy::ProposeNormal);
+        }
+
         if (req.has_header() && req.get_header().get_read_quorum()) ||
-           !self.raft_group.raft.in_lease() || req.get_requests().len() == 0 {
-            return false;
+           !self.raft_group.raft.in_lease() {
+            return Ok(RequestPolicy::ReadIndex);
         }
 
         // If applied index's term is differ from current raft's term, leader transfer
         // must happened, if read locally, we may read old value.
         if self.get_store().applied_index_term != self.raft_group.raft.term {
-            return false;
-        }
-
-        for cmd_req in req.get_requests() {
-            if cmd_req.get_cmd_type() != CmdType::Snap && cmd_req.get_cmd_type() != CmdType::Get {
-                return false;
-            }
+            // TODO: add it in queue directly.
+            return Ok(RequestPolicy::ReadIndex);
```

Comment:
No, this pr's goal is to avoid propose read request. No matter if `commit_index` is equal to current term or not, this is handled properly in `read_index`.